### ISSUE TITLE
XSLT cannot return a document node to eXist-db XQuery

### ISSUE THREAD

**Issue by adamretter** [eXist-db/exist#2096]
Returning an `<xsl:document>` from transform:transform actually causes eXist-db to return an XDM `element()`, it should instead return an XDM `document-node()`.

**Comment by duncdrum** [comment]
document-node handling is repeatedly creating problems it seems.
see #1964, #1932, #1562, #1464, …

**Comment by adamretter** [comment]
Yeah, it seems out in-memory DOM is basically broken and has been for a long time. I think it was based on Saxon's TinyTree, my plan is to bring over the latest version from Saxon as soon as I can.

**Comment by brihaye** [comment]
If I remember well, one of the developers, years ago, commited some code that removed all document-node handling...
